[CS2103T-F12-2] VISIT

gachia opened the pull request:
Made change so that it's possible to do a PR to original repo

jon-chua commented on docs/DeveloperGuide.adoc:
```diff
@@ -301,7 +301,15 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |user |have <<user-defined-macros,user-defined macros>> |streamline my diagnosis documentation
 
 |`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
```
Some comments on your User Stories:

1. The actors of your use cases are pretty general (user), maybe we could change them to something more descriptive and specific? (e.g. doctor)
1. Try to add more user stories! 😄 
1. Great that you linked specific definitions to your glossary! 👍

jon-chua commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+
 _{More to be added}_
 
 [appendix]
```
Some comments on your NFRs:

1. Try to update them soon! For example, we could rename persons to patients to make it more specific 😄 
1. Think of more specific NFRs tied to your application.

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
++
+Use case ends.
+[none]
+* 3. The given name/index is invalid.
```
Numbering should be 3a instead of 3

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+* 3. The given name/index is invalid.
++
+[none]
+** 3a. AddressBook shows an error message.
```
Numbering should be 3a1 instead of 3a

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+Use case resumes at step 2.
+
+[none]
+* 4. The profile is empty.
```
Numbering should be 4a instead of 4

jon-chua reviewed (commented):
Some comments on your Developer Guide were highlighted below! 😸

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to add new visitation record for a specific patient in the list
```
Maybe it will be good to clarify whether the record added to a specific patient will be a custom one everytime, or just an empty one after it is requested by the user.

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+
+Use case resumes at step 1.
+
+* 1b. There are multiple records with the same name and date
```
Will records with the same date and time but not same name be allowed?

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+* 1b. There are multiple records with the same name and date
++
+[none]
+** 1b1. AddressBook shows list of records and prompts user to re-enter last command with index of record to delete.
```
If records with the same name or datetime are not allowed in the records, would this still be an issue?

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Save user-defined macros
```
Might want to consider adding the definition of macro into the glossary

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+*Extensions*
+
+[none]
+* 2. No reminders are set to show
```
Extensions should start with 1 instead of 2

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+*Extensions*
+
+[none]
+* 2. No reminders are scheduled for the day
```
Extensions should start with 1 instead of 2

Raikonen commented on docs/DeveloperGuide.adoc:
```diff
@@ -341,6 +349,212 @@ Use case ends.
+Use case ends.
+
+[discrete]
+=== Use case: View follow-up reminders
```
Might want to consider adding the definition of follow-up reminders, and its difference with ordinary reminders into the glossary too!

Raikonen reviewed (commented):
Overall, the Use Cases were well done and can be improved as more details regarding the application is discussed and decided upon!

Q-gabe committed:
Merge pull request #9 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

Q-gabe committed:
Merge pull request #10 from Q-gabe/master

chore: sync update-UGDG branch with team repo

Q-gabe committed:
docs: update miscellaneous docs to remove AB3 references

Q-gabe committed:
docs: fix broken links and general housekeeping of DeveloperGuide.adoc

Q-gabe committed:
Merge pull request #76 from Q-gabe/update-UGDG

docs: clean-up docs for documentation integrity

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

SQwQ committed:
Merge branch 'master' of https://github.com/SQwQ/main

SQwQ committed:
docs: fixed headings for Design considerations

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #77 from SQwQ/master

fixed heading

SQwQ committed:
Merge branch 'master' of https://github.com/SQwQ/main

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

Wingedevil committed:
feat: remove required flag for unalias

Wingedevil committed:
docs: update ug with new commands

Wingedevil committed:
refactor: refactor alias table accessors and mutators

Wingedevil committed:
docs: edit use cases for alias

Wingedevil committed:
refactor: refactor apply alias

Wingedevil committed:
docs: add implementation of alias into DG

Wingedevil committed:
fix: fix tests

Wingedevil committed:
Merge branch 'master' into branch-alias-command

SQwQ committed:
Merge pull request #80 from Wingedevil/branch-alias-command

Update UG/DG, refactor unalias

ReignOfComputer committed:
docs: follow-up and reminders update

ReignOfComputer committed:
Merge pull request #81 from ReignOfComputer/docs-reminders-1.3

docs: follow-up and reminders update

gachia committed:
Updated DevGuide for UI redesign (#78)

docs: added GUI redesign part in Implementation section of DevGuide
docs: added Ui Redesign Class Diagram PlantUML file and image
cleanup: Rename ProfilePanel to ProfileWindow to follow naming structure
cleanup: added fxid to ReminderPanel for future use

Q-gabe committed:
Merge pull request #11 from AY1920S1-CS2103T-F12-2/master

chore: sync update-UGDG branch with team repo

ReignOfComputer committed:
fix: reminder model declarations

ReignOfComputer committed:
Merge branch 'master' into fix-reminder-model-declaration

ReignOfComputer committed:
Merge pull request #82 from ReignOfComputer/fix-reminder-model-declaration

fix: reminder model declarations

ReignOfComputer committed:
feat: follow-up command stub

Q-gabe committed:
docs: updated DG to include `profile` component

Q-gabe committed:
Merge branch 'master' into update-UGDG

Q-gabe committed:
Merge pull request #84 from Q-gabe/update-UGDG

docs: update DG for `profile` command

ReignOfComputer committed:
feat: follow-up and reminder refactor

ReignOfComputer committed:
Merge branch 'master' into feat-follow-up-stub

ReignOfComputer committed:
fix: unstick Travis

ReignOfComputer committed:
Merge branch 'feat-follow-up-stub' of https://github.com/ReignOfComputer/main into feat-follow-up-stub

ReignOfComputer committed:
fix: Travis Time Travel

ReignOfComputer committed:
Merge pull request #83 from ReignOfComputer/feat-follow-up-stub

feat: follow-up command stub

ReignOfComputer committed:
feat: preliminary support for persistant reminders

gachia committed:
Updated Visit forms and Profile to fit UI redesign (#85)

* Updated Visit forms and Profile to fit UI redesign

chore: VisitRecordForm and VisitListPanel follow Ui design
cleanup: Added Label to VisitListPanel for improved clarity
cleanup: after discussing, changed ProfileWindow to not use pop-up style
cleanup: fix formatting issues with MainWindow
chore: updated CSS for the forms

* cleanup: remove unused import statments

* cleanup: fix checkstyle in css file

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,6 +153,104 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+image::AddAliasSequenceDiagram.png[]
+
+[NOTE]
+The check for and illegal alias consists of creating a new `AddressBookParser` and parsing the given alias into it, which is not shown in this diagram.
```
![image](https://user-images.githubusercontent.com/42912708/67545071-68c5ee80-f72a-11e9-8bd2-cf4bc5df73d0.png)

Unclear phrasing. 😕

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,6 +153,104 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+Step 1. The user executes `profile 3` command view the profile of the 3rd person shown in VISIT's current list. It extracts the `Person` instance from the filtered Person list from the current `model` and returns a `CommandResult` with the `Person` instance as an attribute. `MainWindow#executeCommand()` is called and the check for `CommandResult#isShowProfile()` passes, calling `ProfileWindow#set ...
+
+image::Profile0.png[]
```
Wishful Thinking!

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -241,7 +339,95 @@ image::CommitActivityDiagram.png[]
+
+===== Aspect: Data structure to support the addvisit/deletevisit/editvisit commands
+
+*  Use a arraylist to store the `VisitReport` objects in `VisitList` object
```
Use an arraylist.

Q-gabe replied:
OOPS! :-(

jon-chua commented on docs/diagrams/AddVisit0.puml:
```diff
@@ -0,0 +1,25 @@
+else ([else])
+    :date is set as current date;
+endif
+:shows pop-up form for user to fill in;
+:User saves changes;
+:VisitList is updated by invoking SaveVisitCommand;
```
Maybe we could try to standardize the capitalization of the UML diagrams 😄

jon-chua commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,6 +153,104 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+** Cons: May cause dependency issues as this increases coupling, with relation to the Profile and Patient class. Foreseeably, this may cause issues especially when patient attributes are to be added or removed.
+* **Alternative 2:** Create an explicit `Profile` instance every call.
+** Pros: Decreases coupling by cutting down on the direct calls to the `Person` instance.
+** Cons: We must ensure that the `Profile` instance generated is correct every time.
```
I think that you guys should try to elaborate on your current choice, try to go beyond listing the pros and cons and explain why you have chosen the first alternative over the other 😄

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -241,7 +339,95 @@ image::CommitActivityDiagram.png[]
+If the user executes `deletevisit 1` instead, the index value will be set to `-1` and the `VisitList#deleteRecord()` block would be skipped, but the rest of the flow remains the same.
+
+
+==== Design Considerations
```
Good to explain how you taken into consideration these concerns. And maybe why you did or did not implement them.

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -241,7 +339,95 @@ image::CommitActivityDiagram.png[]
+image::EditVisit0.png[]
+
+[NOTE]
+If the user executes `editvisit 1` instead, a `CommandResult` with a `ObservableList<VisitReport>` object will be returned. `MainWindow#executeCommand()` runs and the check for `CommandResult#isEditVisit()` fails while the check for `CommandResult#isShowVisitList()` passes, calling `VisitListPanel#setup()` which populates the `VisitListPanel` with the contents of the `ObservableList<VisitReport>` ...
```
This **should not be a note**. It is too long. If you would want this to be a note, please make it shorter and simpler to understand.

seanlowjk commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,6 +153,104 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+image::Profile0.png[]
+_{Show activity diagram of opening profile window}_
+
+Step 2. After viewing the relevant information, the user presses the 'esc' key which closes the form. This causes an event handler in `ProfileWindow` which listens for a `KeyEvent.KEY_RELEASED` event where the event code equates to `KeyCode.ESCAPE` ('esc' key) to trigger, which hides the profile window.
```
(`esc` key)? Is this necessary?

fatclarence reviewed (commented):
Use case: Delete Person has yet to be updated (it reflects a deletion from AddressBook which I believe should not be the name of your application.)
Update section F.3 Saving Data and the section below it as well.

In section 3.4.1, I think you could consider changing the commands addvisit and deletevisit to addVisit and deleteVisit because the convention is to camelcase the method names.
Also in 3.4.1, it might be more intuitive when you execute the editvisit command, your isEditVisit should return true, not "CommandResult#isEditVisit() fails" as you stated, because that is quite counter-intuitive. If I want to edit the visit, the isEditVisit should return me "true". Unless that is not the intention of your explanation. If it is not your intention, then I feel it could be explained better. In fact, with your UML diagram, you do not have to explain the UML diagram again in your information because it should be clear enough in your UML diagrams.

jon-chua commented on docs/DeveloperGuide.adoc:
```diff
@@ -241,7 +339,95 @@ image::CommitActivityDiagram.png[]
+
+*  Use a arraylist to store the `VisitReport` objects in `VisitList` object
+** Pros: Fast retrieval by index.
+** Cons: Need to handle index out of bound errors.
```
Try to explain the alternatives besides these design that you have chosen and why your current choice is better than the alternatives!

seanlowjk reviewed (commented):
Hope you guys can take a look at the comments given. Not many as not many sections have been updated. All the best for DG!

jon-chua reviewed (commented):
Some general comments on your developer guide are listed here! 😄

ReignOfComputer committed:
feat: LocalDate Storage

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

SQwQ committed:
fix: fixed bug #86
UI: Changed display text for VisitList

SQwQ committed:
fix: fixed bug where medicine/diagnosis/prescription fields are reset to null everytime add/edit/deletevisit is run after VISIT restarts

SQwQ committed:
fix: SaveVisitCommand's success message displays correctly
UI: Changed display message for addvisit/deletevisit/savevisit commands
UI: VisitListPanel of reports closes before form pop ups for edit.

ReignOfComputer committed:
feat: update days remaining for Reminders the next day

SQwQ committed:
docs: updated UML diagrams for DG

ReignOfComputer committed:
feat: reminders and follow-up 1.3 gold

ReignOfComputer committed:
fix: checkstyle

ReignOfComputer committed:
Merge branch 'master' into feat-reminders-1.3

SQwQ committed:
Docs and bug fixes (#89)

* fix: fixed bug #86
UI: Changed display text for VisitList

* fix: fixed bug where medicine/diagnosis/prescription fields are reset to null everytime add/edit/deletevisit is run after VISIT restarts

* fix: SaveVisitCommand's success message displays correctly
UI: Changed display message for addvisit/deletevisit/savevisit commands
UI: VisitListPanel of reports closes before form pop ups for edit.

* docs: updated UML diagrams for DG

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

ReignOfComputer committed:
Merge branch 'master' into feat-reminders-1.3

ReignOfComputer committed:
Merge pull request #90 from ReignOfComputer/feat-reminders-1.3

feat: reminders and follow-up 1.3 gold

Wingedevil committed:
feat: add command to show all existing aliases

Wingedevil committed:
Merge branch 'master' into branch-alias-command

Wingedevil committed:
fix: fix trailing whitespace on line 143 of UserPrefs.java

gachia committed:
Added Reminder UI to the MainWindow (#93)

* Added Reminder UI to the MainWindow

chore: added the ReminderCard and ReminderList for the MainWindow to display
chore: added ReminderStub object for use
chore: added test arraylist to test functionality in MainWindow in Reminder (suggest changing Reminder to Reminders or something similar)
note: Obtaining Reminder list goes through same flow as Person list. This is through Logic -> Model.

* fix checkstyle

* temp fix: Remove failing test check temporarily

Remove failing test check get the commit out. @ReignOfComputer will improve code logic to ensure the test case passes.

* fix checkstyle.........again

SQwQ committed:
Merge branch 'master' into branch-alias-command

SQwQ committed:
Merge pull request #92 from Wingedevil/branch-alias-command

feat: add command to show all existing aliases

SQwQ committed:
Merge remote-tracking branch 'upstream/master'

SQwQ committed:
docs: fixed some copy-pasted comments
feat: List Empty prompt similar to HelpWindow is shown instead of empty VisitListPanel when chosen person for edit/deletevisit has no past records
chore: changed no index prompt messages for edit/deletevisit

SQwQ committed:
docs: UG

SQwQ committed:
docs: UG
fix: checkstyle

SQwQ committed:
fix: checkstyle

gachia committed:
More UI updates (#94)

* More UI updates

chore: Increase window size for increased visual space
chore: Increase resultDisplay to fit 4 lines
chore: Added header text to fields of PersonCard and VisitCard
chore: Added header for null/empty fields for VisitCard
chore: updated AliasList design
chore: Removed selected effect and added hover effect in CSS
chore: updated UI design in VisitListPanel
chore: updated window titles in VisitListPanel and Alias List

* fix checkstyle

SQwQ committed:
fix: visit list panel will now hide when visit form shows

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #95 from SQwQ/master

UG and UI update

Q-gabe committed:
docs: add DG UML files for `profile` command

Wingedevil committed:
docs: updated user guide

Wingedevil committed:
test: add more test for alias feature

Wingedevil committed:
Merge branch 'master' into branch-alias-command

SQwQ committed:
Merge pull request #96 from Wingedevil/branch-alias-command

Update ug and added more tests

gachia committed:
More, More UI changes (#97)

* More UI updates

chore: added 2 icons to the MainWindow for more visual appeal
chore: added list of quick commands to HelpWindow for users
chore: updated CSS for HelpWindow
docs: updated help command to reflect changes in HelpWindow in User Guide

* fix checkstyle

chore: removed hover over list cell to prevent unintended interaction

Q-gabe committed:
chore: update java doc comments for ProfileWindow.java

Q-gabe committed:
Merge pull request #12 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

Q-gabe committed:
Merge branch 'update-UGDG'

Q-gabe committed:
docs: add `Profile` activity and sequence diagram

Q-gabe committed:
docs: update DG for `profile` section

Q-gabe committed:
docs: update UG to fix format errors

Q-gabe committed:
feat: add new button for profile generation

Q-gabe committed:
feat: add functionality to print Patient Profile from Profile Window

Q-gabe committed:
chore: change final build jar filename to match UG

ReignOfComputer committed:
Merge pull request #102 from Q-gabe/master

feat: add functionality to export `profile` as .txt file, `profile` reflects `visitList`

ReignOfComputer committed:
fix: disallow blank reminders

Closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/91

ReignOfComputer committed:
chore: refactor Reminders to AppointmentTable

ReignOfComputer committed:
feat: hook into Main Window

Closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/103

ReignOfComputer committed:
docs: update Developer Guide for Appointments System

ReignOfComputer committed:
fix: restored nulled tests

ReignOfComputer committed:
Merge pull request #104 from ReignOfComputer/feat-reminders-ui-update

feat: appointments UI update

ReignOfComputer committed:
feat: anti-duplicate check

ReignOfComputer committed:
fix: logic error resulting in repeated appointments

ReignOfComputer committed:
feat: delete appointment stub

ReignOfComputer committed:
feat: delete appointment command

ReignOfComputer committed:
feat: sorting command for appointments

ReignOfComputer committed:
fix: checkstyle

ReignOfComputer committed:
Merge pull request #108 from ReignOfComputer/feat-additional-appointments-magic

feat: final enhancements to Appointments

Q-gabe committed:
Merge pull request #13 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

SQwQ committed:
docs: Updated UG images

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #110 from SQwQ/master

docs: Updated UG images

Q-gabe committed:
feat(ui): enhance appointment panel on MainWindow

Wingedevil committed:
docs: update activity diagram

Wingedevil committed:
Merge branch 'master' into branch-alias-command

Q-gabe committed:
Merge branch 'master' into reminder-panel-rework

Wingedevil committed:
doc: add missing new line

Q-gabe committed:
Merge pull request #111 from Q-gabe/reminder-panel-rework

feat(ui): enhance appointment panel on MainWindow

gachia committed:
Fixed Reminders GUI not updating (#113)

* Fixed Reminders GUI not updating

fix (dirty): Created a AppointmentList class to handle data for GUI usage, while Appointment Table still stores the data in jsonfile. Both run in parallel, which means data is added in both objects when using Reminder Command.
fix: some renaming for better clarity and format
temp: disabled delete and sort reminder command as they are not ready for v1.3
@ReignOfComputer will continue refactoring this.

* Removed DeleteReminder command test

because command is currently disabled

Wingedevil committed:
Merge branch 'master' into branch-alias-command

Q-gabe committed:
Merge pull request #112 from Wingedevil/branch-alias-command

docs: update activity diagram

Q-gabe committed:
Merge pull request #15 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

gachia committed:
Updated UI image (#114)

ReignOfComputer committed:
fix: appointment system parallel updates

ReignOfComputer committed:
chore: major appointment system refactoring

ReignOfComputer committed:
chore: additional appointment system refactoring

ReignOfComputer committed:
chore: appointments test v1

ReignOfComputer committed:
docs: update UG for appointments

ReignOfComputer committed:
docs: update DG for appointments

ReignOfComputer committed:
fix: checkstyle

ReignOfComputer committed:
Merge pull request #116 from ReignOfComputer/fix-appointments-1.3

hotfix: appointment system 1.3 parachute

ReignOfComputer committed:
fix: restrict invalid days for appointments (#118)

ReignOfComputer committed:
chore: detect collisions on AppointmentList for sync with AppointmentTable

ReignOfComputer committed:
feat: sort appointments on app launch

ReignOfComputer committed:
Merge branch 'master' into chore-col-det-appointments-list

Q-gabe committed:
Merge pull request #16 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

ReignOfComputer committed:
Merge pull request #119 from ReignOfComputer/chore-col-det-appointments-list

chore: detect collisions on Appointment to maintain sync

ReignOfComputer committed:
chore: handle invalid input above max int

Closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/131, closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/132.

ReignOfComputer committed:
fix: checkstyle

ReignOfComputer committed:
feat: clear command now clears appointments

SQwQ committed:
fix: #122 #123 #138 #140 #142

SQwQ committed:
Merge pull request #135 from ReignOfComputer/chore-handle-invalid-int

chore: handle invalid input above max int

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #148 from SQwQ/master

fix: #122 #123 #138 #140 #142

SQwQ committed:
fix: closes #141 #152 #153 #154
fixed bug which shows up when the app initialises without a Json file

SQwQ committed:
Merge branch 'master' of https://github.com/SQwQ/main

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #183 from SQwQ/master

 fix: #141 #152 #153 #154

Q-gabe committed:
feat: change `profile` window TextArea to dark theme

Q-gabe committed:
fix: minor java style issue

ReignOfComputer committed:
fix: removeappt required params

Closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/177.

ReignOfComputer committed:
chore: hard-lock arguments

Closes https://github.com/AY1920S1-CS2103T-F12-2/main/issues/163.

ReignOfComputer committed:
Merge pull request #184 from ReignOfComputer/fix-ped

fix: ped

ReignOfComputer committed:
tests: 20191102 checkpoint

ReignOfComputer committed:
fix: checkstyle

ReignOfComputer committed:
Merge pull request #185 from ReignOfComputer/tests-20191102

tests: 20191102 checkpoint

Q-gabe committed:
feat: change `find` command to allow substring search

Q-gabe committed:
feat: additional handlers for 'p' and 'q' key press

Q-gabe committed:
fix (docs): update UG to new `find` behavior

Q-gabe committed:
docs: update UG to implemented `profile` behaviour

Wingedevil committed:
fix: fix alias being applied multiple times

#159

Wingedevil committed:
Merge branch 'branch-alias-command' of https://github.com/Wingedevil/main into branch-alias-command

Wingedevil committed:
fix: fix easter egg

 #146, #126

Wingedevil committed:
fix: aliaslist title not shown in full

#126

Wingedevil committed:
fix: docs: fix ug having an incorrect command as an example

#147

Wingedevil committed:
Merge branch 'master' into branch-alias-command

Wingedevil committed:
Merge pull request #186 from Wingedevil/branch-alias-command

fix: fix #126, #146, #147 and #159

gachia committed:
fix: #182, #178, #165, #130 (#187)

Fixed these issues.
fix: #182
fix: #178
fix: #165
fix: #130

Tags will always be in lower-case and will not have duplicates
Long input fixed in ResultDisplay. It will now wrap text.
Long input in CommandBox is now visible and not blocked by the Send button
All commands are now available in the help window. The user will still be required to see the User Guide for more detailed information.

Q-gabe committed:
fix: fix VisitRecordForm not centering in tabbed view

Q-gabe committed:
fix: fix `profile` not updating by making closing ProfileWindow mandatory

Q-gabe committed:
fix: fix checkstyle in ProfileWindow.java

Q-gabe committed:
test: re-edited `find` related tests to fit new behaviour

Q-gabe committed:
docs: update README.adoc

Q-gabe committed:
docs(fix): add missing emoji

Q-gabe committed:
Merge branch 'master' into master

Q-gabe committed:
Merge pull request #189 from Q-gabe/master

fix!: #180, #167, #150, and #124

Wingedevil committed:
fix: fix aliases not being case insensitive

ReignOfComputer committed:
tests: 20191103 checkpoint

Q-gabe committed:
feat: display Profile Visit as ListView

Q-gabe committed:
fix: fix checkstyle violations

ReignOfComputer committed:
Merge pull request #190 from ReignOfComputer/tests-20191103

tests: 20191103 checkpoint

ReignOfComputer committed:
docs: update UG

ReignOfComputer committed:
docs: update DG

ReignOfComputer committed:
fix: newline

ReignOfComputer committed:
Merge pull request #191 from ReignOfComputer/docs-20191103

docs: update UG and DG

Q-gabe committed:
Merge branch 'master' into profile-extension

ReignOfComputer committed:
chore: ppp

ReignOfComputer committed:
fix: newline

ReignOfComputer committed:
Merge pull request #194 from ReignOfComputer/chore-ppp

chore: ppp

ReignOfComputer committed:
fix: asciidocs includes

GitHub does not support asciidocs includes

ReignOfComputer committed:
chore: longer ppp

ReignOfComputer committed:
Merge pull request #195 from ReignOfComputer/fix-asciidoc-includes

fix: asciidocs includes

SQwQ committed:
Merge branch 'master' into branch-alias-command

SQwQ committed:
Merge pull request #193 from Wingedevil/branch-alias-command

fix: fix aliases not being case insensitive

SQwQ committed:
Merge branch 'master' into profile-extension

Q-gabe committed:
Merge pull request #192 from Q-gabe/profile-extension

feat: `ProfileWindow` now shows Visits as Cells instead of raw text

gachia committed:
More UI fixes (#196)

fix: Fix minor newline error in Help and added Date format example.
fix: Motd window now follows CSS style
fix: Added new icons for the titles, replacing them with unicode characters that change based on platform.
fix: Added package font so that it remains consistent throughout all platforms. Required @Q-gabe to test.

SQwQ committed:
fix: closes #127 Updated error message and years 19xx are accepted too, #175 Exit using esc key, #188
docs: Updated UG,DG and PPP
tests: added and edited tests
refactor: refactor boolean variable names in CommandResult to comply with coding standards

SQwQ committed:
fix:checkstyle

SQwQ committed:
chore:changed back to f2

SQwQ committed:
fix: checkstyle

SQwQ committed:
fix: exit

SQwQ committed:
Merge pull request #197 from SQwQ/master

fix: #127, #175, #176

ReignOfComputer committed:
chore: branding refactor

ReignOfComputer committed:
fix: docs regression from refactoring

ReignOfComputer committed:
Merge pull request #198 from ReignOfComputer/chore-brand-refractor

chore: brand refactor

Q-gabe committed:
Merge pull request #17 from AY1920S1-CS2103T-F12-2/master

chore: sync with team repo

gachia committed:
Fixed UI and updated UGDG (#199)

* Fixed UI and updated UGDG

fix: Save Button in VisitRecordWindow now shows the corresponding button to save
removed: F2 shortcut is removed for Message of the Day since there is already a command to display it.
docs: updated Readme to current implementation
docs: updated AB3 portions of UG to fit current implementation
docs: updated Logic, Model, Storage diagrams and expanded on GUI redesign section in DG

* fix checkstyle

SQwQ committed:
chore: changed PPP name

SQwQ committed:
Merge pull request #201 from SQwQ/master

chore: changed PPP name

SQwQ committed:
fix: fixe broken PPP link

SQwQ committed:
Merge branch 'master' into master

Q-gabe committed:
chore: add generated profiles folder to gitignore

Q-gabe committed:
fix(ui): fix font not displaying on MacOS

SQwQ committed:
Merge pull request #202 from SQwQ/master

fix: fix broken PPP link orz

Q-gabe committed:
Merge branch 'master' into master

gachia committed:
Merge pull request #203 from Q-gabe/master

fix: fix portability issues of Gill Sans MT font

Q-gabe committed:
fix(ui): add box for Visits in Profile Panel

gachia committed:
fix: Packaged Font loaded properly in MainWindow (#204)

* fix: Packaged font loaded correctly in MainWindow

gachia committed:
PPP submission (#205)

* Updated PPP

docs: Wrote personal PPP and added previous Ui Concept design for it.
docs: Updated AboutUs page to reflect new PPP.

ReignOfComputer committed:
fix: force absolute pathing on site image

ReignOfComputer committed:
chore: center images and captions on docs

ReignOfComputer committed:
docs: formatting and cleanup

ReignOfComputer committed:
chore: optimized ppp

ReignOfComputer committed:
fix: newline

ReignOfComputer committed:
Merge pull request #206 from ReignOfComputer/docs-cleanup

docs: multiple changes

gachia committed:
docs: aligned images in PPP to center (#207)

* docs: updated PPP to align images to center

Wingedevil committed:
fix: fix alias only matching fully

Q-gabe committed:
refactor!: refactor `Profile` component to abide coding standards

Q-gabe committed:
test: add uni tests for ProfileCommand, GenerateProfileCommand, StringUtil, ProfileUtil

Q-gabe committed:
Merge branch 'master' into master

Q-gabe committed:
chore(docs): update README.adoc

Q-gabe committed:
chore(docs): update UG for `profile` component for v1.4

SQwQ committed:
refactor: changed magic numbers
fix: fixed bug where editvisit 1 i/-1 will not throw error
fix: fixed the same bug for deletevisit

Q-gabe committed:
Merge pull request #208 from Q-gabe/master

refactor!: refactor of `Profile` Component and Tests

SQwQ committed:
chore: checkstyle compliance
docs:updated UG of new behavior for deletevisit and editvisit command

SQwQ committed:
Merge branch 'master' into master

SQwQ committed:
Merge pull request #209 from SQwQ/master

Refactor

Wingedevil committed:
docs: update UG and DG

Wingedevil committed:
chore: update logger of aliaslist window

Wingedevil committed:
docs: add PPP and update UG

ReignOfComputer committed:
chore: enum appointments

ReignOfComputer committed:
Merge pull request #211 from ReignOfComputer/chore-appointments-enum

chore: enum appointments

Wingedevil committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-F12-2/main into branch-alias-command

Wingedevil committed:
fix: fix build

Wingedevil committed:
fix: import orderof aliaswindow

SQwQ committed:
Merge pull request #210 from Wingedevil/branch-alias-command

docs: update UG and PPP

Q-gabe committed:
docs: add PPP for q-gabe and edit responsibilities on AboutUs.adoc

Q-gabe committed:
docs: update UG formatting for `profile`

Q-gabe committed:
docs: update DG `profile` section for v1.4

Q-gabe committed:
Merge branch 'master' into master

Q-gabe committed:
Merge pull request #212 from Q-gabe/master

docs: add PPP, update UG and DG

Q-gabe committed:
fix: adjust PPP for page limit

Q-gabe committed:
Merge branch 'master' into master

Q-gabe committed:
Merge pull request #213 from Q-gabe/master

fix: adjust PPP for page limit

ReignOfComputer committed:
fix: refactoring artifacts

Oops!

ReignOfComputer committed:
fix: json root name

ReignOfComputer committed:
Merge pull request #214 from ReignOfComputer/fix-refactoring-artifacts

fix: refactoring artifacts

gachia committed:
Final Updates to UGDG (#215)

* Final Updates to UGDG

docs: fix @Wingedevil link to PPP in About Us
docs: Added Manual Testing Appendix to DG
docs: Updated Ui image to reflect latest build
docs: Standardised image sizes in UG
small fix: made label focus untraversable in CommandBox
